jsengine-ios/jsengine/WebRequest.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  WebRequest.swift
+//  jsengine
+//
+//  Created by Ghadir Eraisha on 12/12/16.
+//  Copyright © 2016 Cliqz GmbH. All rights reserved.
+//
+
+import UIKit
+import WebKit
+import JavaScriptCore
+
+class WebRequest {
+    weak var jsContext: JSContext? = nil
+    var tabs = NSMapTable.strongToWeakObjectsMapTable()
+    
+    init() {
+        
+    }
+    
+    func extend(context: JSContext) {
+        self.jsContext = context
+        let webRequest = JSValue.init(newObjectInContext: context)
+        
+        let nativeIsWindowActive: @convention(block) (Int) -> Bool = {[weak self] tabId in
+            if let tabActive = self?.isTabActive(tabId) {
+                return tabActive
+            }
+            return false
+        }
+        webRequest.setObject(unsafeBitCast(nativeIsWindowActive, AnyObject.self), forKeyedSubscript: "_nativeIsWindowActive")
+        
+        context.setObject(webRequest, forKeyedSubscript: "webRequest")
+    }
+    
+    func shouldBlockRequest(request: NSURLRequest) -> Bool {
+        guard BlockedRequestsCache.sharedInstance.hasRequest(request) == false else {
```

Comment:
It's in the current code of Anti-tracking to improve performance. Based on discussion with Mahmoud, he sees that it's not needed anymore. I will update that.